server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -374,11 +377,19 @@ private String getChecksumOfLocalFile(String file) throws IOException {
         return localSegmentChecksumMap.get(file);
     }
 
-    private void deleteStaleCommits() {
+    private void deleteStaleCommitsAsync(String threadpoolName, Runnable onCompletion) {
         try {
-            remoteDirectory.deleteStaleSegments(LAST_N_METADATA_FILES_TO_KEEP);
-        } catch (IOException e) {
-            logger.info("Exception while deleting stale commits from remote segment store, will retry delete post next commit", e);
+            indexShard.getThreadPool().executor(threadpoolName).execute(() -> {
+                try {
+                    remoteDirectory.deleteStaleSegments(LAST_N_METADATA_FILES_TO_KEEP);
+                } catch (IOException e) {
+                    logger.info("Exception while deleting stale commits from remote segment store, will retry delete post next commit", e);
+                }
+                onCompletion.run();
```

Comment:
Move this to finally
```
onCompletion.run();
```

Reply:
Isn't 
```
remoteDirectory.deleteStaleSegments(LAST_N_METADATA_FILES_TO_KEEP);
```
a blocking call, which means the finally would only execute once the delete has succeeded. Am I missing something?